src/service_gobject.rs

```diff
@@ -116,3 +119,25 @@ impl ServiceGObject {
         //self.set_property("trust-level", query_result.trust_level as u8);
     }
 }
+
+impl From<QueryResult> for ServiceGObject {
+    fn from(qr: QueryResult) -> Self {
+        let QueryResult {
+            name,
+            description,
+            status,
+            trust_level,
+            service_type,
+            vm_name,
+            ..
+        } = qr;
```

Comment:
btw, I seen a crate with magic macro to do assignments like this
(ignore it, unless you too curious to try it)